### ISSUE TITLE
Add Android Auto support

### ISSUE THREAD

**Issue by TECHNOFAB11** [home-assistant/android#791]
Inspired by [this](https://www.reddit.com/r/homeassistant/comments/icplnv/running_home_assistant_on_android_auto/) Reddit post, is it possible to add Android Auto functionality?

Developer Guide: https://developer.android.com/training/cars

Problem is that there are only Media and Messaging apps supported, any ideas how to still show the Home Assistant interface? It should probably be less interactive tho, so people don't use it while driving etc. But for now it would suffice if the web interface could be shown, even if only as proof of concept

**Comment by myhrmans** [comment]
Google does not allow that sadly. Hence why I made the guide that I linked in that post. All apps must be allowed by Google and an app like Home Assistant will sadly never be allowed since the possibilities with Home Assistant are endless.

**Comment by TECHNOFAB11** [comment]
Sorry to bother you guys again, but I just wanted to let y'all know that Smarthings works with Android Auto: https://youtu.be/Bk1aT0_PqUk

In my opinion the UI isn't the best and could easily distract the driver. I'm not sure why Google allowed it, but if they allow this then there is a good chance that they'll allow Home Assistant too (you guys are way better at UI than what Smarthings is doing :smiley: )

**Comment by wrkn4alivn** [comment]
See this [blog post](https://android-developers.googleblog.com/2021/04/start-your-engines-launch-new-android.html) and related media coverage.

It seems like there could be a path recently opened up for some Auto functionality for HA. I think there are some use cases that maintain the principle of only allowing driver-relevant interactions, in a minimally distractive way. 

There's no reason to allow unlimited possibilities in the production version of HA Android. An ideal MVP would be a single screen with up to two elements: cover/door (for garage or gate), or camera snapshot.

I would love the Android app even more!

**Comment by katsaplias** [comment]
bump

**Comment by OliverSkivdal** [comment]
Support for only Home Assistant media browser in Android Auto would be amazing if google allows it!

**Comment by dshokouhi** [comment]
Looks like a new article got posted! I guess they will now accept us?

https://developer.android.com/training/cars/apps/iot

**Comment by GSzabados** [comment]
@JBassett, thanks for the answer. And please excuse me for my ignorance, but when is the next beta due? I managed to join beta in the Play Store.

Thanks again!

**Comment by dshokouhi** [comment]
beta gets pushed every Saturday night 9pm PST and then it depends on Play store review, you'll need to have some patience as it may not be available for a few days after the push while we wait for approval.  

In the meantime you should be able to update the APK in the releases tab to take advantage of it today.

**Comment by dshokouhi** [comment]
actually updating the phone app may not work, seems the update needs to be installed via play store so it will show.

https://github.com/home-assistant/android/pull/3211#issuecomment-1380517432

**Comment by GSzabados** [comment]
![20230112_204355](https://user-images.githubusercontent.com/35445496/212165928-5041de81-039a-42ec-863f-f37e47343dbd.jpg)

It is indeed working.

As a bit of feedback. If it would be possible to select some favourites from the app that would be great. I don't know how it sources the lights, locks, etc. But for locks I have actually one lock, but I have all the Child locks from Z2M listed.

So a favourites would make things less cluttered when there are dozens of other lights, switches, etc.

The navigation works, but when there are dozens of options it get a bit messy, what I think it is only Android Auto, as it jumps back continuously to the back arrow on the top left.

Anyhow, it is an amazing addition to the app! Just need to make it easily accessible what would be used from the car and it ticks the safety requirements as well.

**Comment by dshokouhi** [comment]
@GSzabados please do not add feature request to an already closed request, instead open a new one for each individual request. Do not combine requests please, they all need to be separated.

**Comment by ronaldt80** [comment]
How do I install this? You guys are awesome!

**Comment by GSzabados** [comment]
@dshokouhi, no worries. I will open a feature request.

**Comment by dshokouhi** [comment]
> How do I install this? You guys are awesome!

update the app from github repo or join play store beta and wait for review.

**Comment by GSzabados** [comment]
> How do I install this?

If you are browsing GitHub from the phone, got to the main page of this repository, scroll down until you see the releases part, and select the + 12.. releases link.

Or just go here

https://github.com/home-assistant/android/releases

 and download from the last beta release's Assets the full apk and install it.

**Comment by GSzabados** [comment]
@dshokouhi and @JBassett, this is really awesome! Tried in the garage with the light switch there and works.

**Comment by ronaldt80** [comment]
Thanks, installed the beta but nothing in my launcher of android auto. anything else in terms of permissions which need to be changed? Great addition in any event!

**Comment by GSzabados** [comment]
> installed the beta but nothing in my launcher of android auto

Have you started the app and the release notes popped up as well? I had to install it twice. The first time it hasn't updated it for some reason. The second time the beta was installed correctly and when opened the app the release notes popped up as it should be.

**Comment by dreimer1986** [comment]
Worked from the very start and with no problems at all. I have Developer mode enabled on my Android Auto and there "Unknown sources". Likely not needed for this case, but I have a few more gimmicks installed on my Android Auto and thus already used this setting. Maybe that is the difference here between those who can already see the icon and those who do not. But do NOT ask here how to enable developer mode in Android Auto. Just Google it. If you cannot find it in Google then you indeed should wait for the final in Play Store.

Finally no more three times shouting at Google Assistant to open my garage door and only managed playing some random song on Spotify or switch some random light on instead... Two "clicks" and it shows a nice button on the display.

**Comment by ronaldt80** [comment]
Got release notes (very limited) for 2013.1; looking at app version it's the right one. Am (of course 😄 ) on developer mode and unknown sources enabled.

Which android auto version do you have installed? Know in the past certain AA functionality was rolled out in phases (or by geography in view of safety regulations). I'm on 8.5 (released yesterday) but see 8.6 and 8.7beta are also available on apk sites.

**Comment by GSzabados** [comment]
I am on:
Android Auto: 8.6.625034-release
Home Assistant: beta-3062-a5cbe043-full

Developer mode on, but unknown sources off.

**Comment by ronaldt80** [comment]
Don't know what did the trick but after upgrading to 8.7, connecting by usb to the car (vs. wireless) it showed up. Not yet able to open my front door (needs a script) but turning on the lights in the kitchen was a fantastic first step.

Thanks for the support, this is great!

**Comment by pbzdyl** [comment]
I was able to also test the beta version and close the gate/cover from Android Auto - however, it kept showing it's open even though the gate closed and further clicking on the gate icon was doing nothing. Shall I open a separate issue?

**Comment by jpelgrom** [comment]
> I was able to also test the beta version and close the gate/cover from Android Auto - however, it kept showing it's open even though the gate closed and further clicking on the gate icon was doing nothing. Shall I open a separate issue?

If you're able to reproduce this, please open a new issue. The app will basically show the entity's raw state so it'd be helpful if you can provide the state information from developer tools. If you open the app on your phone before starting Android Auto I think you can access the logs on the phone as well.
